client/examples/node_api_core/04_post_block.rs

```diff
@@ -27,7 +28,11 @@ async fn main() -> Result<()> {
     // Get parents for the block.
     let parents = Parents::new(client.get_tips().await?)?;
     // Create the block.
-    let block = Block::build(parents).finish(min_pow_score)?;
+    let block = Block::build(parents).finish_nonce_provider(|bytes| {
```

Comment:
You mean something like this?
```Rust
pub fn do_pow(bytes: &[u8], min_pow_score: u32) -> Result<u64>{
    let miner = MinerBuilder::new().with_num_workers(num_cpus::get()).finish();

    miner.nonce(bytes, min_pow_score)
}
```

Reply:
Yes

Reply:
And what about the other parameters like worker count or the cancel thing? Not needed?
Also I don't expect anyone really to use that, we can also just change the example to `client.block().finish().await?` and it will do the same (actually even better because it will update the tips if PoW takes too long)